src/Network/Ricochet/Types.hs

```diff
@@ -27,10 +30,15 @@ makePacket chan bs = MkPacket (4 + (fromIntegral $ B.length bs)) chan bs
 data Connection = MkConnection
   { _cHandle      :: Handle
   , _cChannels    :: [Channel]
-  , _cIsClient    :: Bool
   , _cInputBuffer :: ByteString
   }
 
+-- | Creates an initial 'Connection' from a Handle
+makeConnection :: Handle -> Connection
+makeConnection handle = -- Start out with only a Control Channel
+  let channels = [MkChannel 0 $ MkChannelType "im.ricochet.control-channel"]
+  in  MkConnection handle channels B.empty
+
 instance Eq Connection where
```

Comment:
First startpage result for “haddock instance comments”: http://trac.haskell.org/haddock/ticket/29. I guess that fix will have made it into the versions we’re using. I remember seeing it in some haddock documentation already.